### PR TITLE
[FIX] account: add onchange warning when setting a contact's company

### PR DESCRIPTION
**Pull request by guva-odoo**
If a partner has already some journal items, we display a warning
when setting the partner's company.

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by guva-odoo** [comment]
Check `journal_item_count` instead of `supplier_rank`/`customer_rank` because a partner can have a rank without having a supplier/customer rank
https://github.com/odoo/odoo/blob/9947e82c5c25cde030aca43f7eb6fe1293300329/addons/account/models/partner.py#L542-L552

**Comment by guva-odoo** [comment]
related to https://github.com/odoo/enterprise/pull/24553

**Comment by Levizar** [comment]
@william-andre I guess we should do this then

File: addons/account/models/partner.py [hunk 1]
```diff
@@ -489,6 +489,18 @@ def _onchange_company_id(self):
             company = self.env.company
         return {'domain': {'property_account_position_id': [('company_id', 'in', [company.id, False])]}}
 
+    @api.onchange('parent_id')
+    def onchange_parent_id(self):
+        result = super().onchange_parent_id()
+        if self.sudo()._origin.journal_item_count:
```
Comment (william-andre) [inline addons/account/models/partner.py:495]: But `journal_item_count` is not a stored field, which was the only reason to check `customer_rank`. Even worse, it is computed by doing a  `search_count` in a loop.

I don't think that that cost is worth for this issue.

**Review by william-andre — CHANGES_REQUESTED** [review]

File: addons/account/models/partner.py (hunk 1 above)
Comment (guva-odoo) [reply addons/account/models/partner.py:495]: What about checking invoice_ids ?

**Comment by C3POdoo** [comment]
Dear @guva-odoo,

Thank you for your contribution but the version 13.0 is no longer supported.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense for the upper version, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**